src/forms/CurrentExpenses.js

```diff
@@ -298,272 +418,219 @@ const Housing = function ({ current, type, time, updateClientValue }) {
- *     household member?
+/** Renders a yes/no choice that will reveal the cash
+ *     flow component given when the user selects 'yes'.
+ *
+ *     We added this extra step between the user and
+ *     the input because people kept skipping that question.
+ *
+ * @param {object} props
+ * @param {bool} props.hasExpenses True if client has any
+ *     expenses here that could affect their income.
+ * @param {object} props.CashFlowRow To be rendered if user
+ *     chooses 'yes'.
+ * @param {string | object} props.label To be rendered as
+ *     the yes/no question.
+ * @param {object} props.propData Data for the prop changed
+ *     by the given cash flow component. (move component in
+ *     here?)
+ * @param {string} props.propData.childPropName Name of cash
+ *     flow client prop to be updated.
+ * @param {object} props.propData client Current or future
+ *     client data.
+ * @param {function} props.propData update Updates client
+ *     values
+ *
+ * @returns Value that React can render
  */
+const EarnedFrom = function ({ hasExpenses, CashFlowRow, label, propData }) {
 
-// `props` is a cloned version of the original props. References broken.
-/**
-  * @function
-  * @param {object} props
-  * @param {function} props.updateClientValue Setting client state
-  * @param {object} props.navData Buttons for bottom row
-  * @param {object} props.client Data for calculating benefits, `future` and `current`
-  * @param {object} props.snippets  Language-specific content
-  *
-  * @returns React element
-  */
-const CurrentExpensesStep = function ({ updateClientValue, navData, client, snippets }) {
+  /** @todo Save amount temporarily when 'source'
+   *      amount is set to 0. */
+  var reset = function (evnt) {
+    var { childPropName, update } = propData;
 
-  return (
-    <FormPartsContainer
-      title     = { snippets.i_title }
-      clarifier = { snippets.i_clarifier }
-      navData   = { navData }
-      formClass = { `expenses` }>
-      <ExpensesFormContent
-        updateClientValue = { updateClientValue }
-        current={ client.current }
-        time={ 'current' }
-        snippets={ snippets } />
-    </FormPartsContainer>
-  );
+    update(evnt, {
+      name:  childPropName,
+      value: 0,
+    });
+  };
+
+  if (hasExpenses) {
+
+    var { childPropName, client } = propData;
+    var showProps = {
+      childName:           childPropName,
+      showChildrenAtStart: client[ childPropName ] > 0,
+      question:            label,
+      heading:             null,
+      onNo:                reset,
+      // `<Surrounder>` props
+      Left:                <AttentionArrow />,
+    };
 
-};  // End CurrentExpensesStep()
+    return (
+      <div className= { 'earned-from' }>
+        <ShowOnYes { ...showProps }>
+          { CashFlowRow }
+        </ShowOnYes>
+      </div>
+    );
 
+  } else {
+    return null;
+  }
 
-export { CurrentExpensesStep };
+};  // Ends <EarnedFrom>
+
+
+export {
```

Comment:
Is any other file actually making use of any of these other components? Do we need to export anything other than the top level component?

Reply:
@todo

Reply:
I would test the exported component only; if we use `mount`, we should be able to find the necessary components. These seem like internal components; essentially an implementation detail that should not be part of the module's API.

Reply:
Hmm, maybe worth a proper discussion. Been considering moving the logic of whether to show stuff or not into the individual components themselves and that would make stuff easier to test individually, etc.